.github/workflows/go-tests.yml

```diff
@@ -0,0 +1,142 @@
+name: go-tests
+on:
+  pull_request:
+  push:
+
+jobs:
+  # FIXME: We already lint in integration.yml, which should we keep?
```

Comment:
> FIXME: We already lint in integration.yml, do we need to be linting twice? Which should we keep? 

I'm thinking we can remove the one from the `integration.yml` and keep this which is doing unit testing anyway.

Reply:
I removed the integration.yml linting in favor of this workflow